s3stream/src/main/java/com/automq/stream/utils/threads/EventLoop.java

```diff
@@ -79,6 +89,9 @@ public void execute(Runnable task) {
 
     public CompletableFuture<Void> shutdownGracefully() {
         shutdown = true;
+        if (!shutdownCf.isDone()) {
```

Comment:
To ensure concurrent safety, the operations of modifying `shutdown` and `tasks` need to be safeguarded by a lock. Similarly, other methods that use these two variables need to be modified.

Reply:
> To ensure concurrent safety, the operations of modifying `shutdown` and `tasks` need to be safeguarded by a lock. Similarly, other methods that use these two variables need to be modified.

Yes, perhaps we can modify the shutdown from a constant to an AtomicBoolean, utilizing CAS (Compare-And-Swap) to ensure concurrent safety.

Reply:
> To ensure concurrent safety, the operations of modifying `shutdown` and `tasks` need to be safeguarded by a lock. Similarly, other methods that use these two variables need to be modified.

now, the tasks variable has also been modified.